octoprint_psucontrol/__init__.py

```diff
@@ -545,16 +549,25 @@ def turn_psu_on(self):
             if self.sensingMethod not in ('GPIO','SYSTEM'):
                 self._noSensing_isPSUOn = True
          
-            time.sleep(0.1 + self.postOnDelay)
+            threading.Timer(0.1 + self.postOnDelay, self.after_psu_on_delay).start()
+
 
+    def after_psu_on_delay(self):
             if self.afterOnGCodeCommands:
                 for command in self._afterOnGCodeCommandsArray:
                     self._printer.commands(command)
 
+            if self.postOnConnect:
+                self._printer.connect()

```

Comment:
This won't work, since it tries to send commands to the printer before connecting to it.

Reply:
Good point, this should be the other way around. Why even try to send GCODE without switching the printer on before?

Reply:
Fixed…
https://github.com/kantlivelong/OctoPrint-PSUControl/pull/148/commits/eb655c388aa32526c07da041430b717a40b4461b